Tools/px_uploader.py

```diff
@@ -869,6 +873,11 @@ def main():
                     # print the error
                     print("\nERROR: %s" % ex.args)
 
+                except FirmwareNotSuitableException:
+                    unsuitable_board = True
+                    up.close()
+                    continue
```

Comment:
Why do you `continue` here and then check `unsuitable_board` instead of just doing `sys.exit(2)` right here?

Reply:
Because there could be other ports attached with a matching board id.

Reply:
Aha!